src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.ResultData;
+import core.basesyntax.service.OperationStrategy;
+import core.basesyntax.service.ShopService;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public List<ResultData> process(List<FruitTransaction> transactions) {
+        Map<String, Integer> resultDataMap = new HashMap<>();
```

Comment:
this map represents DB, so better make it a public static field in the Storage class